backup-stores/testkit/src/main/java/io/camunda/zeebe/backup/testkit/SavingBackup.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.backup.testkit;
+
+import io.camunda.zeebe.backup.api.Backup;
+import io.camunda.zeebe.backup.api.BackupStore;
+import io.camunda.zeebe.backup.testkit.support.TestBackupProvider;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.time.Duration;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ArgumentsSource;
+
+public interface SavingBackup {
+  BackupStore getStore();
+
+  @ParameterizedTest
+  @ArgumentsSource(TestBackupProvider.class)
+  default void savingBackupIsSuccessful(final Backup backup) {
+    Assertions.assertThat(getStore().save(backup)).succeedsWithin(Duration.ofSeconds(10));
+  }
+
+  @ParameterizedTest
+  @ArgumentsSource(TestBackupProvider.class)
+  default void backupFailsIfBackupAlreadyExists(final Backup backup) {
+    // when
+    getStore().save(backup).join();
+
+    // then
+    Assertions.assertThat(getStore().save(backup)).failsWithin(Duration.ofSeconds(10));
+    // TODO
+    //  .withThrowableOfType(Throwable.class);
+    //  .withRootCauseInstanceOf(BackupInInvalidStateException.class);
```

Comment:
❓ guess the TODO is here as this depends on a specific class from the s3 module?

would add an interface method to obtain the expected class here, wdyt?
`Class<? extends Exception> getBackupInInvalidStateExceptionClass();`

resulting in 
```
        .withThrowableOfType(Throwable.class)
        .withRootCauseInstanceOf(getBackupInInvalidStateExceptionClass());
```

Reply:
addressed with https://github.com/camunda/zeebe/pull/10269/commits/fb64966633d2245f24913036631773dc2a5b97f7